spec/requests/api/products_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe "Api::Products", type: :request do
+  
+  describe "Create Product" do
+    let(:product_params) do
+      {
+        name: "test product",
+        code: "PROD01"
+      }
+    end
+
+    context "For Success" do
+      before do
+        post "/api/products/create", params: product_params
+      end
+      
+      it "returns http status success" do
+        expect(response).to have_http_status(201)
+      end
+
+      it "returns succees message" do
+        json_response = JSON.parse(response.body)
+        expect(json_response['message']).to eq("Product Created Successfully")
+      end
+    end
+
+    context "For name validation failure" do
+      before do
+        product_params[:name] = nil
+        post "/api/products/create", params: product_params
+      end
+
+      it "should return failure status" do
+        expect(response).to have_http_status(406)
+      end
+
+      it "should have failure message" do
+        json_response = JSON.parse(response.body)
+        expect(json_response['message']).to eq("Failed To Create Product")
+      end
+
+      it "should contain errors" do
+        json_response = JSON.parse(response.body)
+        expect(json_response['errors']).not_to be_nil
+      end
+    end
+
+    context "For code validation failure" do
+      before do
+        product_params[:code] = nil
+        post "/api/products/create", params: product_params
+      end
+
+      it "should return failure status" do
```

Comment:
Could you please try to use https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples here so that this and spec#34 won't get repeated (DRY).

Reply:
Added shared examples to achive DRY